### PR TITLE
fix: fix the driver queryset

### LINKED ISSUE #1072: `queryset` passed from the `flow.search` API is not parsed properly
**Describe the bug**
In the following codes, `lookups` will be kept as a `google.protobuf.struct_pb2.Struct` which can't be further evaluated in the `apply_all` function

```
        qs = jina_pb2.QueryLang(name='FilterQL', priority=100)
        qs.parameters['lookups'] = {'tags__label': 'label1'}
        query_flow.search(docs[:1], queryset=qs, output_fn=validate_only_label2, callback_on_body=True)
```
reference:
https://github.com/jina-ai/jina/blob/master/jina/drivers/__init__.py#L86
https://github.com/jina-ai/jina/blob/master/jina/drivers/querylang/filter.py#L43

**Describe how you solve it**

---

**Environment**

**Screenshots**

### PR DESCRIPTION
**Pull request by nan-wang**
resolved #1072

The main issue is that the `Struct` is not parsed properly. In this PR, we convert it into `dict`.

### CODE REVIEW

File: tests/integration/issues/github_1072/test_queryset.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from jina.proto import jina_pb2
+from jina.drivers.helper import array2pb
+from jina.flow import Flow
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def test_queryset_with_struct(tmpdir):
+    os.environ['JINA_TEST_INCREMENTAL_INDEX_WORKSPACE'] = str(tmpdir)
```
Comment (JoanFM) [inline tests/integration/issues/github_1072/test_queryset.py:12]: Change env vaiable name for this test?

File: jina/drivers/querylang/filter.py [hunk 2]
```diff
@@ -37,10 +37,12 @@ def __init__(self, lookups: Dict[str, Any], *args, **kwargs):
         self._lookups = Q(**lookups) if lookups else None
```
Comment (hanxiao) [inline jina/drivers/querylang/filter.py:37]:
```suggestion
        self._lookups = lookups
```

I think `Q` builder can be delayed to `__apply_all`

File: jina/drivers/querylang/filter.py [hunk 3]
```diff
@@ -37,10 +37,12 @@ def __init__(self, lookups: Dict[str, Any], *args, **kwargs):
...
 
     def _apply_all(self, docs: Iterable['jina_pb2.Document'], *args, **kwargs) -> None:
+
         if self.lookups:
+            _lookups = Q(**self.lookups) if isinstance(self.lookups, Dict) else self.lookups
```
Comment (hanxiao) [inline jina/drivers/querylang/filter.py:42]:
```suggestion
            _lookups = Q(**self.lookups)
```

File: tests/integration/issues/github_1072/test_queryset.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+from jina.proto import jina_pb2
+from jina.drivers.helper import array2pb
+from jina.flow import Flow
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def test_queryset_with_struct(tmpdir):
+    os.environ['JINA_TEST_QUERYSET_WORKSPACE'] = str(tmpdir)
```
Comment (bwanglzu) [inline tests/integration/issues/github_1072/test_queryset.py:12]: one suggestion, you can reuse the fixture what we have in `conftest.py`, to create a new fixture there, something like:
```python
@pytest.fixture(scope='function')
def random_workspace(tmpdir, random_workspace_name):
    os.environ[random_workspace_name] = tmpdir
    yield random_workspace_name
    del os.environ[random_workspace_name]
```
Then you can make use of this fixture everywhere without manually setup test workspace and delete environment variable manually. Kinda like setUp and tearDown in unittest.

Comment (nan-wang) [reply tests/integration/issues/github_1072/test_queryset.py:14]: Good point!

**Review by JoanFM — APPROVED** [review]

**Pull request merged**